data_layer/weavevm/weavevm.go

```diff
@@ -0,0 +1,141 @@
+package weavevm
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/dymensionxyz/roller/cmd/consts"
+	"github.com/dymensionxyz/roller/utils/errorhandling"
+	"github.com/dymensionxyz/roller/utils/keys"
+	"github.com/dymensionxyz/roller/utils/roller"
+	"github.com/pterm/pterm"
+)
+
+const (
+	ConfigFileName               = "weavevm.toml"
+	mnemonicEntropySize          = 256
+	keyringNetworkID      uint16 = 42
+	requiredAVL                  = 1
+	DefaultTestnetChainID        = 9496
+)
+
+type WeaveVM struct {
+	Root        string
+	PrivateKey  string
+	RpcEndpoint string
+	ChainID     uint32
+}
+
+func (w *WeaveVM) GetPrivateKey() (string, error) {
+	return w.PrivateKey, nil
+}
+
+func (w *WeaveVM) SetMetricsEndpoint(endpoint string) {
+}
+
+func NewWeaveVM(root string) *WeaveVM {
+	var daNetwork string
+
+	rollerData, err := roller.LoadConfig(root)
+	errorhandling.PrettifyErrorIfExists(err)
+
+	cfgPath := GetCfgFilePath(root)
+	weavevmConfig, err := loadConfigFromTOML(cfgPath)
+
+	if err != nil {
+		if rollerData.HubData.Environment == "mainnet" {
+			daNetwork = string(consts.WeaveVMMainnet)
+		} else {
+			daNetwork = string(consts.WeaveVMTestnet)
+		}
+
+		weavevmConfig.PrivateKey, _ = pterm.DefaultInteractiveTextInput.WithDefaultText(
+			"> Enter your PrivateKey without 0x",
+		).Show()
+
+		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+			WithDefaultText(
+				"press 'y' when the wallet are funded",
+			).Show()
+
```

Comment:
needs to have retry with balance check

Reply:
Do u have any idea for it? Im feeling a bit stuck

Reply:
something similar that we already have for balance checks:

https://github.com/dymensionxyz/roller/blob/main/cmd/eibc/init/init.go#L161-L189

Reply:
I mean how to query balance of weavevm?